src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -409,8 +411,14 @@ private void RecordPresenceOfBadAttributes(ImmutableArray<CSharpAttributeData> b
                             }
 
                             var attributesToBind = attributeDeclarationSyntax.Attributes;
-                            syntaxBuilder.AddRange(attributesToBind);
-                            attributesToBindCount += attributesToBind.Count;
```

Comment:
>attributesToBindCount += attributesToBind.Count; [](start = 28, length = 48)

It is probably worth keeping these two statements as an optimization for the most common scenario when attributeMatchesOpt is null.